cloudsync/youtube_test.py

```diff
@@ -0,0 +1,169 @@
+    youtube_mocker = mocker.patch('cloudsync.youtube.build')
+    youtube_mocker().captions.return_value.update.return_value.next_chunk.return_value = (None, caption_response)
+    response = YouTubeApi().upload_caption(subtitle, caption_id)
+    assert response == caption_response
+
+
+def test_delete_video(mocker):
+    """
+    Test that the 'delete_video' method executes a YouTube API deletion request and returns the status code
+    """
+    youtube_mocker = mocker.patch('cloudsync.youtube.build')
+    youtube_mocker().videos.return_value.delete.return_value.execute.return_value = 204
+    assert YouTubeApi().delete_video('foo') == 204
+
+
+def test_list_captions(mocker):
+    """
+    Test that the 'list_captions' method executes a YouTube API request and returns a dict with correct values
+    """
+    key = 'en'
+    value = 'Srnr982VEC79QzEBGcBOL_UFmu9U2e-JgOw-EWIxJXEB5Bjltl3Yvg='
+    youtube_mocker = mocker.patch('cloudsync.youtube.build')
+    youtube_mocker().captions.return_value.list.return_value.execute.return_value = {
+        'etag': 'foo',
+        'items': [{
+            'id': value,
+            'kind': 'youtube#caption',
+            'snippet': {
+                'audioTrackType': 'unknown',
+                'language': key,
+                'lastUpdated': '2017-11-15T14:53:21.839Z',
+                'name': 'English',
+                'videoId': '3h-0mkTVbRg'
+            }
+        }],
+        'kind': 'youtube#captionListResponse'
+    }
+    assert YouTubeApi().list_captions('foo') == {key: value}
+
+
+def test_delete_caption(mocker):
+    """
+    Test that the 'delete_caption' method executes a YouTube API deletion request and returns the status code
+    """
+    youtube_mocker = mocker.patch('cloudsync.youtube.build')
+    youtube_mocker().captions.return_value.delete.return_value.execute.return_value = 204
+    assert YouTubeApi().delete_caption('foo') == 204
+
+
+def test_video_status(mocker):
+    """
+    Test that the 'video_status' method returns the correct value from the API response
+    """
+    expected_status = 'processed'
+    youtube_mocker = mocker.patch('cloudsync.youtube.build')
+    youtube_mocker().videos.return_value.list.return_value.execute.return_value = {
+        'etag': '"ld9biNPKjAjgjV7EZ4EKeEGrhao/Lf7oS5V-Gjw0XHBBKFJRpn60z3w"',
+        'items': [{
+            'etag': '"ld9biNPKjAjgjV7EZ4EKeEGrhao/-UL82wRXbq3YJiMZuZpqCWKoq6Q"',
+            'id': 'wAjoqsZng_M',
+            'kind': 'youtube#video',
+            'status': {
+                'embeddable': True,
+                'license': 'youtube',
+                'privacyStatus': 'unlisted',
+                'publicStatsViewable': True,
+                'uploadStatus': expected_status
+            }
+        }],
+        'kind': 'youtube#videoListResponse',
+        'pageInfo': {'resultsPerPage': 1, 'totalResults': 1}}
+    assert YouTubeApi().video_status('foo') == expected_status
```

Comment:
Can you assert that `foo` is an argument in `list`?

Reply:
OK

Reply:
I think you missed this one, can you do something like `youtube_mocker().video_status.assert_called_once_with('foo')`?

Reply:
added